connectors/es/sink.py

```diff
@@ -660,29 +663,50 @@ async def prepare_content_index(self, index, *, mappings=None):
         exists = await self.client.indices.exists(
             index=index, expand_wildcards=expand_wildcards
         )
+
+        mappings = Mappings.default_text_fields_mappings(is_connectors_index=True)
+
         if exists:
+            # Update the index mappings if needed
             self._logger.debug(f"{index} exists")
-            response = await self.client.indices.get_mapping(
-                index=index, expand_wildcards=expand_wildcards
+            await self._ensure_content_index_mappings(index, mappings, expand_wildcards)
+        else:
+            # Create a new index
+            self._logger.info("Attempt to create a new index")
+            await self._create_content_index(
+                index=index, language_code=language_code, mappings=mappings
             )
-            existing_mappings = response[index].get("mappings", {})
-            if len(existing_mappings) == 0 and mappings:
-                self._logger.debug(
-                    "Index %s has no mappings or it's empty. Adding mappings...", index
-                )
-                await self.client.indices.put_mapping(
-                    index=index,
-                    dynamic=mappings.get("dynamic", False),
-                    dynamic_templates=mappings.get("dynamic_templates", []),
-                    properties=mappings.get("properties", {}),
-                    expand_wildcards=expand_wildcards,
-                )
-                self._logger.debug("Index %s mappings added", index)
-            else:
-                self._logger.debug("Index %s already has mappings. Skipping...", index)
-            return
+            self._logger.info(f"Index {index} has been successfully created.")
+
+        return
+
+    async def _ensure_content_index_mappings(self, index, mappings, expand_wildcards):
+        response = await self.client.indices.get_mapping(
+            index=index, expand_wildcards=expand_wildcards
+        )
+
+        existing_mappings = response[index].get("mappings", {})
+        if len(existing_mappings) == 0 and mappings:
+            self._logger.debug(
+                "Index %s has no mappings or it's empty. Adding mappings...", index
+            )
+            await self.client.indices.put_mapping(
+                index=index,
+                dynamic=mappings.get("dynamic", False),
+                dynamic_templates=mappings.get("dynamic_templates", []),
+                properties=mappings.get("properties", {}),
+                expand_wildcards=expand_wildcards,
+            )
+            self._logger.debug("Index %s mappings added", index)
         else:
-            raise IndexMissing(f"Index {index} does not exist!")
+            self._logger.debug("Index %s already has mappings. Skipping...", index)
```

Comment:
I see you mentioned
https://github.com/elastic/connectors-python/blob/6b700bcdbc80e0344dc4b809db9491657df0e3fa/connectors/es/sink.py#L670

Do you want to handle mapping changes here? In normal cases, the index should be created by either Kibana or Connector service. If we find a discrepancy, should we 1) create a new index with the correct mapping or 2) raise an error or 3) simply log a warning message and continue?

Reply:
Good question:

If the connector can't apply mappings to the index it should fail immediately. This can already happen, see this [issue](https://github.com/elastic/enterprise-search-team/issues/5765). 

I guess it would be nice to handle it and pass a human-readable error. Is this what you also meant?